modules/ximgproc/include/opencv2/ximgproc/fourier_descriptors.hpp

```diff
@@ -0,0 +1,133 @@
+            /** @brief Fit two closed curves using fourier descriptors. More details in @cite PersoonFu1977 and @cite BergerRaghunathan1998
+
+            @param ctr number of Fourier descriptors equal to number of contour points after resampling.
+            @param fd Contour defining second shape (Target).
+            @sa ContourFitting
+            */
+            ContourFitting(int ctr=1024,int fd=16):ctrSize(ctr),fdSize(fd){};
+            /** @brief Fit two closed curves using fourier descriptors. More details in @cite PersoonFu1977 and @cite BergerRaghunathan1998
+
+            @param src Contour defining first shape.
+            @param dst Contour defining second shape (Target).
+            @param _alphaPhiST : alpha=_alphaPhiST(0,0), phi=_alphaPhiST(0,1), s=_alphaPhiST(0,2), Tx=_alphaPhiST(0,3), Ty=_alphaPhiST(0,4)
+            @param dist distance between src and dst after matching.
+            @param fdContour false then src and dst are contours and true src and dst are fourier descriptors.
+            @sa estimateTransformation
+            */
+            void estimateTransformation(InputArray src, InputArray dst, OutputArray _alphaPhiST, double *dist=0,bool fdContour=false);
+            /** @brief set number of Fourier descriptors used in estimateTransformation
+
+            @param n number of Fourier descriptors equal to number of contour points after resampling.
+            @sa setCtrSize
+            */
+            void setCtrSize(int n);
+            /** @brief set number of Fourier descriptors when estimateTransformation used vector<Point>
+
+            @param n number of fourier descriptors used for optimal curve matching.
+            @sa setFDSize
+            */
+            void setFDSize(int n);
+            /**
+            @returns number of fourier descriptors
+            @sa getCtrSize
+            */
+            int getCtrSize() { return ctrSize; };
+            /**
+            @returns number of fourier descriptors used for optimal curve matching
+            @sa getFDSize
+            */
+            int getFDSize() { return fdSize; };
+        };
+        //! @addtogroup ximgproc_filters
+        //! @{
+
+        /**
+        * @brief   Fourier descriptors for planed closed curves
+        *
+        * For more details about this implementation, please see @cite PersoonFu1977
+        *
+        * @param   _src   contour type vector<Point> , vector<Point2f>  or vector<Point2d>
+        * @param   _dst   Mat of type CV_64FC2 and nbElt rows A VERIFIER
+        * @param   nbElt number of rows in _dst or getOptimalDFTSize rows if nbElt=-1
+        * @param   nbFD number of FD return in _dst _dst = [FD(1...nbFD/2) FD(nbFD/2-nbElt+1...:nbElt)]
+        *
+        @sa fourierDescriptor
+        */
+        CV_EXPORTS void fourierDescriptor(InputArray _src, OutputArray _dst, int nbElt=-1,int nbFD=-1);
+        CV_EXPORTS void transform(InputArray _src, Mat t,OutputArray _dst, bool fdContour=true);
```

Comment:
Why does this function not documented? And please use Input or Output array instead of `Mat t` here.
You can also wrap these functions into python replacing `CV_EXPORTS` with `CV_EXPORTS_W`